src/Containers/FileUpload/FileUpload.tsx

```diff
@@ -331,247 +153,349 @@ export const FileUpload: React.FC<IFileUploadProps> = ({
-                dispatch({
-                    type: SET_POSITION_TOP_LOADING,
-                    value:
-                        loadingContainerRef.current.getBoundingClientRect()
-                            .top - MARGIN,
-                });
+    const onWorkerMessage = useCallback(
+        (e: any) => {
+            const { base64StringFile, name } = e.data;
+            if (base64StringFile === undefined) {
+                return;
             }
-        }
-        // this is to calculate and set the width of the success and failure container component
-        if (isSuccess || isFailure) {
-            const width = containerRef.current?.getBoundingClientRect().width;
-            if (width) {
-                dispatch({
-                    type: SET_IS_SUCCESS_WIDTH,
-                    value: width - PADDING * 2 - MARGIN * 2,
-                });
-            }
-        }
-        // this sets height of the component, is used to transition between heights.
-        if (isUploading || isSuccess || isFailure) {
-            if (state.totalHeightPlus) {
-                dispatch({ type: SET_HEIGHT, value: state.totalHeightPlus });
-            } else {
-                dispatch({ type: SET_INITIAL_HEIGHT_PLUS_VALUES });
+            const value = informativePanelsState.values.find(
+                (value_) => value_.name === name,
+            );
+            if(value){
+                value.worker.terminate();
+                if (base64StringFile === NO_BASE64STRINGFILE || isTestIsFailure) {
+                    setInformativePanelsState((prev) => ({
+                        ...prev,
+                        values: prev.values.map((value_) => {
+                            if (value_.name === value.name)
+                                return {
+                                    ...value_,
+                                    isSuccess: {
+                                        ...value_.isSuccess,
+                                        value: false,
+                                    },
+                                    isFailure: {
+                                        ...value_.isFailure,
+                                        value: true,
+                                    },
+                                    isUploading: {
+                                        ...value_.isUploading,
+                                        value: false,
+                                    },
+                                };
+                            return value_;
+                        }),
+                        makeItDisappear:[...prev.makeItDisappear,value.name]
+                    }));
+                    
+                } else {
+                    doWithBase64StringFile(base64StringFile);
+                    setInformativePanelsState((prev) => ({
+                        ...prev,
+                        values: prev.values.map((value_) => {
```

Comment:
value_ is an unclear name

Reply:
it's because there exists already a value in scope. both value and value_  are of the same type (IValue)

Reply:
So is one the previous value? And old value. Etc